src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -128,6 +128,35 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void CheckForUnsupportedArgs(Execution::Context& context)
+    {
+        const auto& unsupportedArgs = context.Get<Execution::Data::Installer>().value().UnsupportedArguments;
+        bool unsupportedArgFound = false;
+
+        if (!unsupportedArgs.empty())
+        {
+            for (auto unsupportedArg : unsupportedArgs)
+            {
+                if (unsupportedArg == UnsupportedArgumentEnum::Log && context.Args.Contains(Execution::Args::Type::Log))
+                {
+                    unsupportedArgFound = true;
+                    context.Reporter.Error() << Resource::String::LogArgumentNotSupported << std::endl;
+                }
+
+                if (unsupportedArg == UnsupportedArgumentEnum::Location && context.Args.Contains(Execution::Args::Type::InstallLocation))
+                {
+                    unsupportedArgFound = true;
+                    context.Reporter.Error() << Resource::String::LocationArgumentNotSupported << std::endl;
+                }
+            }
+        }
+
+        if (unsupportedArgFound)
+        {
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_UNSUPPORTED_ARGUMENT);

```

Comment:
Is there a way to make this a soft error?

As a made up scenario to justify it: I want to install 10 apps and I have a script that does `winget install $(App) --log logs\$(App).log` for each one. I don't want it to fail because I didn't know one of them didn't support it